### PR TITLE
Small fixes to CI and deprecate Python 2.7

### PR DESCRIPTION
**Pull request by basnijholt**

### CODE REVIEW

**Comment by amccaugh** [comment]
I see you beat me to my questions already!

**Comment by basnijholt** [comment]
@amccaugh, you see the builds appear here, also you find them in the "Actions" tab next to "Pull requests".

**Comment by basnijholt** [comment]
It's great that all the CI tests finish in under 1 minute BTW!

**Comment by amccaugh** [comment]
Very nice, I'm a big fan of all of the CI being contained within GitHub.  I've gone ahead and deprecated version 2.7, I want to review a few other miscellaneous changes to the code and in the next day or two I'll pull everything into master and release a new version.  Thanks again for your help

**Comment by basnijholt** [comment]
Great!

We might also consider deprecating 3.5 and even 3.6: https://endoflife.date/python

**Comment by amccaugh** [comment]
Hmm, looks like I may have to figure out what's going wrong with 3.8 / 3.9 pytest, I'll take a look at that later today

**Comment by amccaugh** [comment]
I'd like to avoid deprecating 3.5/3.6 until absolutely necessary -- in my experience, non-computational scientists tend to lag behind the general programming populace in terms of updating their software 😄

**Comment by basnijholt** [comment]
Check these changes https://github.com/basnijholt/phidl/commit/9f3d69e75d01ff46323f3a464f01aac36f66f212 which `pyupgrade` automatically makes when upgrading to Python≥3.7.

**Comment by basnijholt** [comment]
I did add `--py3-plus` to pyupgrade and pushed the changes to the branch of this PR.

File: tests/test_path.py [hunk 1]
```diff
@@ -24,7 +24,10 @@ def test_path_extrude_simplify():
     P = pp.arc(radius=10, angle=90, num_pts=300)
     D = P.extrude(width=1, layer=3, simplify=1e-1)
     h = D.hash_geometry(precision=1e-4)
-    assert h == "92a41fb5afa37cde3b06b521ff14d8445d962069"
+    if sys.version_info >= (3, 8):
+        assert h == "eb36d790b478134fa14a08f43fe28f112839ab2d"
+    else:
+        assert h == "92a41fb5afa37cde3b06b521ff14d8445d962069"
```
Comment (basnijholt) [inline tests/test_path.py:30]: @amccaugh, I have added this check which makes the tests pass. The Polygons seem to be the same, it's just that the upstream hashes seem the have changed.

File: tests/test_path.py [hunk 2]
```diff
@@ -99,7 +102,10 @@ def test_path_arc():
 def test_path_spiral():
     P = pp.spiral(num_turns=5, gap=1, inner_gap=2, num_pts=10000)
     h = P.hash_geometry(precision=1e-4)
-    assert h == "1d46b2f8ab45f35ce781083755df41609d2ee628"
+    if sys.version_info >= (3, 8):
+        assert h == "14ca1131f89458c5716fab3c7c7d7ef6b2d52820"
+    else:
+        assert h == "1d46b2f8ab45f35ce781083755df41609d2ee628"
```
Comment (basnijholt) [inline tests/test_path.py:108]: Similar change here.

**Pull request merged**